File: android/app/src/main/java/com/rafaeltmbr/stopwatch/platform/presentation/mappers/impl/ViewTimeMapperImpl.kt
package com.rafaeltmbr.stopwatch.platform.presentation.mappers.impl

import com.rafaeltmbr.stopwatch.platform.presentation.entities.ViewTime
import com.rafaeltmbr.stopwatch.platform.presentation.mappers.ViewTimeMapper
import kotlin.math.max
import kotlin.math.round

class ViewTimeMapperImpl : ViewTimeMapper {
    override fun map(milliseconds: Long): ViewTime {
        val time = max(milliseconds, 0L)

        val fraction = "%02d".format((round(time / 10f).toInt() % 100).toInt())
        val seconds = "%02d".format((time / 1000 % 60).toInt())
        val minutes = "%02d".format((time / 60_000).toInt())

        return ViewTime(
            minutes = minutes.split("").filter { it.isNotEmpty() },
            seconds = seconds.split("").filter { it.isNotEmpty() },
            fraction = fraction.split("").filter { it.isNotEmpty() }
        )
    }
}

File: ios/Stopwatch/platform/presentation/mappers/impl/ViewTimeMapperImpl.swift
class ViewTimeMapperImpl: ViewTimeMapper {
    func map(_ milliseconds: Int) -> ViewTime {
        let time = max(0, milliseconds)
        
        let fraction = String(format: "%02d", Int((Double(time) / 10.0).rounded()) % 100).split(separator: "").map { "\($0)"}
        let seconds = String(format: "%02d", (time / 1_000) % 60).split(separator: "").map { "\($0)"}
        let minutes = String(format: "%02d", (time / 60_000)).split(separator: "").map { "\($0)" }
        
        return ViewTime(
            minutes: minutes,
            seconds: seconds,
            fraction: fraction
        )
    }
}
